File: hadoop-definitive-guide/hdg_ch02_mapreduce/src/main/scala/com/arv/maxtemp/scala/MaxTemperatureReducer.scala
package com.arv.maxtemp.scala

import java.lang.Iterable

import org.apache.hadoop.io.{IntWritable, Text}
import org.apache.hadoop.mapreduce.Reducer

/**
  * Created by arvind on 7/3/17.
  */


class MaxTemperatureReducerScala extends Reducer[Text, IntWritable, Text, IntWritable] {

  override def reduce(key: Text, values: Iterable[IntWritable], context: Reducer[Text, IntWritable, Text, IntWritable]#Context): Unit = {
    var max = 0;
    while (values.iterator().hasNext) {
      val value: IntWritable = values.iterator().next()
      max = if (max > value.get()) max else value.get()
    }
    context.write(key, new IntWritable(max))
  }

}

File: hadoop-definitive-guide/hdg_ch02_mapreduce/src/main/java/com/arv/maxtemp/MaxTemperatureReducer.java
package com.arv.maxtemp;


import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

import java.io.IOException;

/**
 * Created by arvind on 7/3/17.
 */


class MaxTemperatureReducer extends Reducer<Text, IntWritable, Text, IntWritable> {

    @Override

    protected void reduce(Text key, Iterable<IntWritable> values, Context context) throws IOException, InterruptedException {
        int max = 0;
        for (IntWritable value : values) {
            if (max < value.get())
                max = value.get();
        }
        context.write(key, new IntWritable(max));
    }

}